Add connector documentation (#659)
wso2/docs-choreo-dev: en/docs/references/use-a-connector-in-your-component.md

# Use a Connector 

## Use a connector in your component design

> When you design your components in the low-code view, you can search and add the connectors displayed in the Marketplace as follows:

Author: Added a description.

> When you design your components in the low-code view, you can search and add the connectors displayed in the Marketplace as follows:

Reviewer: Suggested some minor changes to it.

> When you design your components in the low-code view, you can search and add the connectors displayed in the Marketplace as follows:

Author: Cannot see the description

> 1. Open the component you are designing in the VS Code Editor and add a connector.

Author: Yes. corrected.

    
    ![Add connector](../assets/img/marketplace/add-connector.png){.cInlineImage-small}

2. Search for the connector you need to add (e.g., World Bank), and click on it to add it.

    ![Select connector](../assets/img/marketplace/select-connector.png){.cInlineImage-small}

If the connector is not available by default, you need to visit the vendor portal, make a payment or provide your Freemium account details if required, and access the required authorization tokens beforehand.

e.g., If you need to use the Azure SQL DB connector, create an Azure account and obtain an authorization token.

## Update a connector

If you want to introduce a change to your component and update the connector you published for it, you need to explicitly republish the connector.

### Update a private connector

To update a private connector, you can republish the same connector version as follows:

1. Sign in to the Choreo Console at [https://console.choreo.dev](https://console.choreo.dev).
2. Navigate to your component and click  **Manage**.
3. Click **Lifecycle** from the left pane.
4. Click **Republish Connector**.

> ### Update a public connector 

Reviewer 2: Let's change this to update a custom connector with 'Public visibility'

Author: If you look at APIs, we don't generally say APIs with public visibility, rather we use the term public APIs. Therefore used the same pattern here too.

Public connectors are immutable. Therefore, we cannot modify and republish the already published connector version. You have to create a new component version to be able to publish a new connector version from it as follows:

> 1. Create a new git branch for the new connector patch version in your GitHub repo. If you are using a Choreo-managed repo, follow the steps below:

Reviewer: Let's use "GitHub" instead of "git" and "repository" instead of "repo".

    1.  Sign in to the Choreo Console at https://console.choreo.dev. 
    2. Navigate to your component and click **Overview** from the left pane. 
    3. Click the **Edit Code** button and navigate to the online code editor to create a new branch through the terminal using the following commands: 
        ```
            git checkout -b <branch-name>
            git push origin <branch-name>
        ```
2. On the Choreo console, click **Deploy** on the left pane. 
3. Expand the **API version** dropdown and click **+ Create New**. 
4. Select the branch you created in step 1 as the **Associated Branch** and provide a new version, for example, 1.0.x. Click **Create** to create a new patch version of the component.

> 4. Continue to deploy component by clicking **Deploy** on the **Build Area** pane.

Reviewer: Change "deploy component" to "deploy the component".

# Use a Connector 

> A connectors is a component that you or another party have developed and exposed as an API in Choreo, which you published as a connector in the [Marketplace](../marketplace.md). A connector lets you share your APIs privately or publicly, promoting reuse of APIs. Developers can use them when they design their components in the low-code view.

Reviewer: ```suggestion
A connector is a component that you or another party have developed and exposed as an API in Choreo and then published as a connector in the [Marketplace]. A connector lets you share your APIs privately or publicly, promoting the reuse of APIs. Developers can use them when they design their components in the low-code view.
```

## Use a connector in your component design
When you design your components in the low-code view, you can search and add the connectors displayed in the Marketplace as follows:

> 1. Open the component you are designing in the web editor and add a connector.

Reviewer: "Web Editor" should be in title caps.

Author: should it? since its not a specific component, should we have it in caps?
